sys_util/src/struct_util.rs

```diff
@@ -25,7 +35,8 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// * `out` - The struct to fill with data read from `f`.
 pub unsafe fn read_struct<T: Copy, F: Read>(f: &mut F, out: &mut T) -> Result<()> {
     let out_slice = std::slice::from_raw_parts_mut(out as *mut T as *mut u8, mem::size_of::<T>());
-    f.read_exact(out_slice).map_err(|_| Error::ReadStruct)?;
+    f.read_exact(out_slice)
+        .map_err(|e| Error::ReadStruct(IoError::new(ErrorKind::Other, format!("{}", e))))?;
```

Comment:
Shouldn't the mapping work out of the box with: ```.map_err(Error::ReadStruct)?;```?

Reply:
Sorry, you are right. I had thought original error type should be unsure, and I really mistook. It should be:

```rust
f.read_exact(out_slice).map_err(|e| Error::ReadStruct(e))?;
```

Reply:
It is updated.